pkg/testing/matrix/matrix.go

```diff
@@ -0,0 +1,457 @@
+				//yarn install
+				cmd := exec.Command("yarn", "install")
+				cmd.Dir = sdkDir
+				err = cmd.Run()
+				assert.NoError(t, err)
+
+				//yarn run tsc
+				cmd = exec.Command("yarn", "run", "tsc")
+				cmd.Dir = sdkDir
+				err = cmd.Run()
+				assert.NoError(t, err)
+
+				//cp ../../README.md ../../LICENSE package.json yarn.lock ./bin/
+				cmd = exec.Command("cp", "../../README.md", "../../LICENSE", "package.json", "yarn.lock", "./bin/")
+				cmd.Dir = sdkDir
+				err = cmd.Run()
+				assert.NoError(t, err)
+
+				//sed -i.bak -e "s/\$${VERSION}/$(VERSION)/g" ./bin/package.json
+				replace := fmt.Sprintf("s/$${VERSION}/%s/g", pkg.Version)
+				cmd = exec.Command("sed", "-i.bak", "-e", replace, "./bin/package.json")
+				cmd.Dir = sdkDir
+				err = cmd.Run()
+				assert.NoError(t, err)
+			}
+		}
+	}
+
+	//Instantiate new directories and run pulumi convert on each language with new directories as output.
+	for _, langOpt := range opts.Languages {
+		var projectGenerator projectGeneratorFunc
+		switch langOpt.Language {
+		case DOTNET:
+			projectGenerator = dotnetgen.GenerateProject
+		case GO:
+			projectGenerator = gogen.GenerateProject
+		case NODEJS:
+			projectGenerator = jsgen.GenerateProject
+		case PYTHON:
+			projectGenerator = pygen.GenerateProject
+		case JAVA:
+			projectGenerator = func(directory string, project workspace.Project,
+				p *pcl.Program, localProjects map[string]string) error {
+				return javagen.GenerateProject(directory, project, p)
+			}
+		case YAML: // nolint: goconst
+			projectGenerator = func(directory string, project workspace.Project,
+				p *pcl.Program, localProjects map[string]string) error {
+				return yamlgen.GenerateProject(directory, project, p)
+			}
+		default:
+			assert.FailNow(t, "Unsupported language: "+langOpt.Language)
+		}
+
+		subdir := filepath.Join(dir, langOpt.Language)
+		//check if subdir exists
+		if _, err := os.Stat(subdir); !os.IsNotExist(err) {
+			assert.NoError(t, os.RemoveAll(subdir))
+		}
+
+		//create subdir
+		err = os.MkdirAll(subdir, 0755)
+		assert.NoError(t, err)
+
+		//generate project
+		err = projectGenerator(subdir, *proj, pclProgram, localProjects)
+		assert.NoError(t, err)
+
+		//Configure opts
+		langPtOpts := i.ProgramTestOptions{}
+		if opts.Program != nil {
+			langPtOpts = *opts.Program
+		}
+		if langOpt.Opts != nil {
+			langPtOpts = langPtOpts.With(*langOpt.Opts)
+		}
+		langPtOpts = langPtOpts.With(i.ProgramTestOptions{
+			Dir: subdir,
+		})
+		t.Run(langOpt.Language, func(t *testing.T) {
+			i.ProgramTest(t, &langPtOpts)
+		})
+		//clean up subdir
+
+		//Including this seems to cause all tests to fail.
+		/*t.Cleanup(func() {
+			os.RemoveAll(subdir)
+		})*/
+	}
+}
+
+func PythonConfigurePkg(pkg *schema.Package) error {
+	pkg.Version = &semver.Version{
+		Major: pkg.Version.Major,
+		Minor: pkg.Version.Minor,
+		Patch: pkg.Version.Patch,
+	} // Prune patch and build versions, since semver and python don't seem to agree on formatting.
```

Comment:
Hah they don't, playing with pulumictl for this over at https://github.com/pulumi/pulumictl/pull/51